### PR TITLE
mLrnQueue has encoded as a class

### LINKED ISSUE #6645: 2.13a1: doInBackgroundAnswerCard - RuntimeException
2.13a1 bugs out with "Collection not opened" for two of my four active decks (plus their parent):

Previous version was whatever had been on the Playstore alpha channel (I finally joined).

```
25574 I AnkiDroid: Obtaining card
21479 I AnkiDroid: AnkiActivity::onStart - Reviewer
25574 E AnkiDroid: CollectionTask/ doInBackgroundAnswerCard - RuntimeException on answering card
25574 E AnkiDroid: com.ichi2.libanki.WrongId:  No card with id 0
25574 E AnkiDroid:     at com.ichi2.libanki.Card.load(Card.java:24)
25574 E AnkiDroid:     at com.ichi2.libanki.Card.<init>(Card.java:8)
25574 E AnkiDroid:     at com.ichi2.libanki.Collection.getCard(Collection.java:1)
25574 E AnkiDroid:     at com.ichi2.libanki.sched.SchedV2._getLrnCard(SchedV2.java:8)
25574 E AnkiDroid:     at com.ichi2.libanki.sched.SchedV2._getLrnCard(SchedV2.java:1)
25574 E AnkiDroid:     at com.ichi2.libanki.sched.SchedV2._getCard(SchedV2.java:1)
25574 E AnkiDroid:     at com.ichi2.libanki.sched.SchedV2.getCard(SchedV2.java:4)
25574 E AnkiDroid:     at com.ichi2.async.CollectionTask.doInBackgroundAnswerCard(CollectionTask.java:10)
25574 E AnkiDroid:     at com.ichi2.async.CollectionTask.actualDoInBackground(CollectionTask.java:51)
25574 E AnkiDroid:     at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:2)
25574 E AnkiDroid:     at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:1)
25574 E AnkiDroid:     at android.os.AsyncTask$3.call(AsyncTask.java:378)
25574 E AnkiDroid:     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
25574 E AnkiDroid:     at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
25574 E AnkiDroid:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
25574 E AnkiDroid:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
25574 E AnkiDroid:     at java.lang.Thread.run(Thread.java:919)
25574 E AnkiDroid:
25574 E AnkiDroid: com.ichi2.libanki.WrongId:  No card with id 0
25574 E AnkiDroid:     at com.ichi2.libanki.Card.load(Card.java:24)
25574 E AnkiDroid:     at com.ichi2.libanki.Card.<init>(Card.java:8)
25574 E AnkiDroid:     at com.ichi2.libanki.Collection.getCard(Collection.java:1)
25574 E AnkiDroid:     at com.ichi2.libanki.sched.SchedV2._getLrnCard(SchedV2.java:8)
25574 E AnkiDroid:     at com.ichi2.libanki.sched.SchedV2._getLrnCard(SchedV2.java:1)
25574 E AnkiDroid:     at com.ichi2.libanki.sched.SchedV2._getCard(SchedV2.java:1)
25574 E AnkiDroid:     at com.ichi2.libanki.sched.SchedV2.getCard(SchedV2.java:4)
25574 E AnkiDroid:     at com.ichi2.async.CollectionTask.doInBackgroundAnswerCard(CollectionTask.java:10)
25574 E AnkiDroid:     at com.ichi2.async.CollectionTask.actualDoInBackground(CollectionTask.java:51)
25574 E AnkiDroid:     at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:2)
25574 E AnkiDroid:     at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:1)
25574 E AnkiDroid:     at android.os.AsyncTask$3.call(AsyncTask.java:378)
25574 E AnkiDroid:     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
25574 E AnkiDroid:     at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
25574 E AnkiDroid:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
25574 E AnkiDroid:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
25574 E AnkiDroid:     at java.lang.Thread.run(Thread.java:919)
```
and further
```
21479 I AnkiDroid: Handling Activity Result: 901. Result: 203
21479 I AnkiDroid: Creating notification channel with id/name: General Notifications/AnkiDroid
21479 I AnkiDroid: Creating notification channel with id/name: Synchronization/Synchronization
21479 I AnkiDroid: Creating notification channel with id/name: Global Reminders/Cards due
21479 I AnkiDroid: Creating notification channel with id/name: Deck Reminders/Reminders
21479 I AnkiDroid: Displaying Database Error
21479 W System.err: java.io.IOException: Cannot run program "sqlite3": error=2, No such file or directory
21479 W System.err:    at java.lang.ProcessBuilder.start(ProcessBuilder.java:1050)
21479 W System.err:    at java.lang.Runtime.exec(Runtime.java:698)
21479 W System.err:    at java.lang.Runtime.exec(Runtime.java:528)
21479 W System.err:    at java.lang.Runtime.exec(Runtime.java:425)
21479 W System.err:    at com.ichi2.anki.dialogs.DatabaseErrorDialog.onCreateDialog(DatabaseErrorDialog.java:8)
21479 W System.err:    at com.ichi2.anki.dialogs.DatabaseErrorDialog.onCreateDialog(DatabaseErrorDialog.java:1)
21479 W System.err:    at androidx.fragment.app.DialogFragment.onGetLayoutInflater(DialogFragment.java:3)
21479 W System.err:    at androidx.fragment.app.Fragment.performGetLayoutInflater(Fragment.java:1)
21479 W System.err:    at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:70)
21479 W System.err:    at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:9)
21479 W System.err:    at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:166)
21479 W System.err:    at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:25)
21479 W System.err:    at androidx.fragment.app.FragmentManagerImpl.executeOps(FragmentManagerImpl.java:6)
21479 W System.err:    at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManagerImpl.java:14)
21479 W System.err:    at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManagerImpl.java:10)
21479 W System.err:    at androidx.fragment.app.FragmentManagerImpl.execPendingActions(FragmentManagerImpl.java:4)
21479 W System.err:    at androidx.fragment.app.FragmentManagerImpl.executePendingTransactions(FragmentManagerImpl.java:1)
21479 W System.err:    at com.ichi2.anki.AnkiActivity.showDialogFragment(AnkiActivity.java:6)
21479 W System.err:    at com.ichi2.anki.AnkiActivity.showAsyncDialogFragment(AnkiActivity.java:2)
21479 W System.err:    at com.ichi2.anki.AnkiActivity.showAsyncDialogFragment(AnkiActivity.java:1)
21479 W System.err:    at com.ichi2.anki.DeckPicker.showDatabaseErrorDialog(DeckPicker.java:2)
21479 W System.err:    at com.ichi2.anki.DeckPicker.handleDbError(DeckPicker.java:2)
21479 W System.err:    at com.ichi2.anki.DeckPicker.onActivityResult(DeckPicker.java:3)
21479 W System.err:    at android.app.Activity.dispatchActivityResult(Activity.java:8145)
21479 W System.err:    at android.app.ActivityThread.deliverResults(ActivityThread.java:4966)
21479 W System.err:    at android.app.ActivityThread.handleSendResult(ActivityThread.java:5014)
21479 W System.err:    at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51)
21479 W System.err:    at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
21479 W System.err:    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
21479 W System.err:    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2131)
21479 W System.err:    at android.os.Handler.dispatchMessage(Handler.java:107)
21479 W System.err:    at android.os.Looper.loop(Looper.java:214)
21479 W System.err:    at android.app.ActivityThread.main(ActivityThread.java:7707)
21479 W System.err:    at java.lang.reflect.Method.invoke(Native Method)
21479 W System.err:    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
21479 W System.err:    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
21479 W System.err: Caused by: java.io.IOException: error=2, No such file or directory
21479 W System.err:    at java.lang.UNIXProcess.forkAndExec(Native Method)
21479 W System.err:    at java.lang.UNIXProcess.<init>(UNIXProcess.java:133)
21479 W System.err:    at java.lang.ProcessImpl.start(ProcessImpl.java:141)
21479 W System.err:    at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
21479 W System.err:    ... 35 more
```

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
## Purpose / Description
I've found where the missing argument of https://github.com/ankidroid/Anki-Android/pull/6637 went

## Fixes
Fixes #6645

## Approach
Creating a class to encode this specific data, I was forced to see where the error is.

## How Has This Been Tested?

Not yet

## Learning (optional, can help others)

Splitting working PR into atomic PR does not work well

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java [hunk 1]
```diff
@@ -891,15 +911,10 @@ protected boolean _fillLrn() {
                             "SELECT due, id FROM cards WHERE did IN " + _deckLimit() + " AND queue IN (" + Consts.QUEUE_TYPE_LRN + ", " + Consts.QUEUE_TYPE_PREVIEW + ") AND due < ?"
                             + " AND id != ? LIMIT ?", new Object[] { cutoff, currentCardId(), mReportLimit});
             while (cur.moveToNext()) {
-                mLrnQueue.add(new long[] { cur.getLong(0), currentCardId(), cur.getLong(1) });
+                mLrnQueue.add(new LrnCard(cur.getLong(0), cur.getLong(1)));
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java:914]: Just to note: bugFix is here.

**Comment by david-allison-1** [comment]
I'll pull and test now

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java (hunk 1 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java:914]: Want I split in two commits ?

**Comment by Arthur-Milchior** [comment]
Thanks.

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java (hunk 1 above)
Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java:912]: Would be useful for anyone in the future

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java:912]: Done

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java [hunk 2]
```diff
@@ -889,17 +907,12 @@ protected boolean _fillLrn() {
                     .getDatabase()
                     .query(
                             "SELECT due, id FROM cards WHERE did IN " + _deckLimit() + " AND queue IN (" + Consts.QUEUE_TYPE_LRN + ", " + Consts.QUEUE_TYPE_PREVIEW + ") AND due < ?"
-                            + " AND id != ? LIMIT ?", new Object[] { cutoff, currentCardId(), mReportLimit});
+                            + " AND id != ? LIMIT ?", new Object[] { cutoff, mReportLimit});
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java:910]: You have 3 substitutions, but only use 2.

**Comment by Arthur-Milchior** [comment]
Corrected. I was making the exact same bug I corrected previously.

I'll really need to put more tests

**Pull request merged**